src/Umbraco.Cms.Targets/buildTransitive/Umbraco.Cms.Targets.targets

```diff
@@ -1,20 +1,46 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Target Name="CopyAppsettingsSchema" BeforeTargets="Build">
+  <!-- Automatically nest files (this needs to be in this file, because the props file is imported too early) -->
+  <ItemGroup>
+    <Content Update="appsettings.*.json">
+      <DependentUpon>appsettings.json</DependentUpon>
+    </Content>
+    <Content Update="appsettings-schema.*.json">
+      <DependentUpon>appsettings-schema.json</DependentUpon>
+    </Content>
+  </ItemGroup>
+  
+  <!-- Copy JSON schema files into the project directory -->
+  <Target Name="CopyUmbracoJsonSchemaFiles" BeforeTargets="Build">
+    <Message Text="Copying JSON schema files into project directory: @(UmbracoJsonSchemaFiles->'%(Filename)%(Extension)')" Importance="high" />
+    <Copy SourceFiles="@(UmbracoJsonSchemaFiles)" DestinationFolder="$(MSBuildProjectDirectory)" SkipUnchangedFiles="true" />
+  </Target>
+
+  <!-- Remove copied JSON schema files from the project directory -->
+  <Target Name="RemoveUmbracoJsonSchemaFiles" BeforeTargets="Clean">
+    <Message Text="Removing JSON schema files from project directory: @(UmbracoJsonSchemaFiles->'%(Filename)%(Extension)')" Importance="high" />
+    <Delete Files="@(UmbracoJsonSchemaFiles->'$(MSBuildProjectDirectory)\%(Filename)%(Extension)')" />
+  </Target>
+
+  <!-- Add references to the JSON schema in the project directory -->
+  <UsingTask TaskName="JsonSchemaAddReferences" AssemblyFile="$(MSBuildThisFileDirectory)..\tasks\netstandard2.0\Umbraco.JsonSchema.Extensions.dll" />
+  <Target Name="AddUmbracoJsonSchemaReferences" BeforeTargets="Build" DependsOnTargets="CopyUmbracoJsonSchemaFiles">
     <ItemGroup>
-      <_SchemaFiles Include="$(MSBuildThisFileDirectory)..\appsettings-schema.json" />
-      <_SchemaFiles Include="$(MSBuildThisFileDirectory)..\appsettings-schema.Umbraco.Cms.json" />
+      <!-- Include references to JSON schema files that are copied into the project directory (if they aren't already referenced and didn't opt-out) -->
+      <UmbracoJsonSchemaReferences Include="@(UmbracoJsonSchemaFiles->'%(Filename)%(Extension)#')" Exclude="@(UmbracoJsonSchemaReferences)" Condition="'%(UmbracoJsonSchemaFiles.Reference)' != 'false'" />
```

Comment:
Just a quick note on why the reference is postfixed with a `#`: although VS Code loaded the schemas without issues, Visual Studio 2022 seems to require this. I'm still trying to fully understand the JSON schema format, but this should make it point to the root of the schema, which is technically correct...